src/core/reference.js

```diff
@@ -62,55 +86,146 @@
  */
 
 /**
+ * Creates and names a <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions">function</a>.
+ * A <a href="#/p5/function">function</a> is a set of statements that perform a task.
+ * Optionally, functions can have parameters. <a href="https://developer.mozilla.org/en-US/docs/Glossary/Parameter">Parameters</a>
+ * are variables that are scoped to the function, that can be assigned a value when calling the function.
+ *
+ * From <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function">the MDN entry</a>:
  * Declares a function with the specified parameters.
  * @property function
  *
  * @example
  * <div class='norender'>
  * <code>
- * function calRectArea(width, height) {
+ * function calculateRectArea(width, height) {
  *   return width * height;
  * }
- * calRectArea(4, 5); //calling the function
+ * calculateRectArea(4, 5); // calling the function
  * </code>
  * </div>
  */
 
 /**
  * Specifies the value to be returned by a function.
+ * For more info checkout <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/return">
+ * the MDN entry for return</a>.
  * @property return
  *
  * @example
  * <div class='norender'>
  * <code>
- * function calSquare(x) {
+ * function calculateSquare(x) {
  *   return x * x;
  * }
- * calSquare(4); //return 16
+ * calculateSquare(4); // returns 16
+ * </code>
+ * </div>
+ */
+
+/**
+ * @submodule Types
+ */
+
+/**
+ * A <a href="#/p5/boolean">boolean</a> is one of the 7 <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Primitive_values">primitive data types</a> in Javascript.
+ * A boolean can only be `true` or `false`.
+ *
+ * From <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Boolean_type">the MDN entry</a>:
+ * Boolean represents a logical entity and can have two values: true, and false.
+ *
+ * @property boolean
+ *
+ * @example
+ * <div class='norender'>
+ * <code>
+ * const myBoolean = false;
```

Comment:
For all of the types examples, maybe use `let` per @limzykenneth's comment above.